CHANGELOG.md

```diff
@@ -1,5 +1,12 @@
 # Figbird Changelog
 
+## 0.17.0
+
+- Introduce support for endpoints that do not respond with `result.total` or `result.total` as -1 when using `allPages: true` to indicate that the endpoint is not counting the total available results, instead figbird will keep paging over the API until all data is gathered. Use `optimisticParallelLimit` to control how many requests to trigger when using in conjunction with `parallel`.
```

Comment:
Wondering if `optimisticParallelLimit` is actually useful? Would we really want to set a different parallel limit if we don't know how many total pages there are?

Reply:
My thinking was roughly that:

* if we know the total number of pages, it's good to for a bit more, e.g. 4 (right now we fetch them all at once, but I think that's also bad, anything without limits is bad..)
* if we don't the total number of pages, we can either fetch one at a time, or optimistically launch some requests ahead of time risking they'll come back blank, but in this case maybe 4 is too much, so it's 2
* by separating those 2 settings, we can handle both kind of endpoints optimally in the same application

Reply:
* and even if the application only has one kind of endpoints (e.g. with total only, or without total only), then the defaults are better for both cases without you having to configure anything

Reply:
sure, sounds good 👍